src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,332 @@
+        item.resourcesPerPatient = resourcesPerPatient[item.patientId];
+        return item;
+    });
+
+    const sortedData = _.orderBy(data, 'resourcesPerPatient', 'desc');
+    return {
+        totalItems: sortedData.length,
+        data: _.values(sortedData),
+    };
+}
+
+@observer
+export class FilesAndLinks extends React.Component<IFilesLinksTable, {}> {
+    constructor(props: IFilesLinksTable) {
+        super(props);
+        makeObservable(this);
+    }
+
+    getDefaultColumnConfig(
+        key: string,
+        columnName: string,
+        isNumber?: boolean
+    ) {
+        return {
+            name: columnName || '',
+            headerRender: (data: string) => (
+                <span data-test={data}>{data}</span>
+            ),
+            render: (data: { [id: string]: string }) => {
+                if (isUrl(data[key])) {
+                    return (
+                        <a href={data[key]} target="_blank">
+                            {data[key]}
+                        </a>
+                    );
+                }
+                return <span data-test={data[key]}>{data[key]}</span>;
+            },
+            download: (data: { [id: string]: string }) => data[key] || '',
+            sortBy: (data: { [id: string]: any }) => {
+                if (data[key]) {
+                    if (isNumber) {
+                        return parseFloat(data[key]);
+                    } else {
+                        return data[key];
+                    }
+                }
+                return null;
+            },
+            filter: (
+                data: { [id: string]: string },
+                filterString: string,
+                filterStringUpper: string
+            ) => {
+                if (data[key]) {
+                    if (!isNumber) {
+                        return (data[key] || '')
+                            .toUpperCase()
+                            .includes(filterStringUpper);
+                    }
+                }
+
+                return false;
+            },
+        };
+    }
+
+    @observable searchTerm: string | undefined = undefined;
+
+    readonly getData = remoteData({
+        await: () => [
+            this.props.store.selectedSamples,
+            this.props.store.resourceDefinitions,
+            this.props.store.sampleResourceData,
+        ],
+        onError: () => {},
+        invoke: async () => {
+            if (this.props.store.selectedSamples.result.length === 0) {
+                return Promise.resolve({ totalItems: 0, data: [] });
+            }
+            const sampleData = await fetchFilesLinksData(
+                this.props.store.filters,
+                this.props.store.sampleResourceData.result!,
+                this.searchTerm,
+                'patientId',
+                'asc',
+                RECORD_LIMIT
+            );
+
+            return Promise.resolve(sampleData);
+        },
+    });
+
+    readonly columns = remoteData({
```

Comment:
Ideally this should be `@computed` not `remoteData`

```suggestion
    @computed get columns() {
```

Here is a sample implementation https://github.com/cBioPortal/cbioportal-frontend/blob/0b095b1c3eef6d2b4119acb36795720202b231a2/src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx#L142

Reply:
Done